torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -620,174 +750,128 @@ class IrParser {
-            const float kEps = eps.value();
+              auto b = sum(grad_x_hat, outer_reduction_axes);
+              auto bcast_b = broadcast(b, outer_broadcast_mask);
 
-            const size_t kNormShapeNumDims = norm_shape->vec().size();
-            const size_t kOuterNumDims = input->nDims() - kNormShapeNumDims;
+              auto c1 = mul(grad_x_hat, x_hat);
+              auto c2 = sum(c1, outer_reduction_axes);
+              auto bcast_c2 = broadcast(c2, outer_broadcast_mask);
+              auto c3 = mul(x_hat, bcast_c2);
 
-            std::vector<int> outer_reduction_axes(kOuterNumDims);
-            std::vector<bool> outer_broadcast_mask(input->nDims(), false);
-            for (size_t idx = 0; idx < kOuterNumDims; ++idx) {
-              outer_reduction_axes[idx] = idx;
-              outer_broadcast_mask[idx] = true;
-            }
+              auto inner = sub(sub(a, bcast_b), c3);
 
-            std::vector<int> inner_reduction_axes(kNormShapeNumDims);
-            std::vector<bool> inner_broadcast_mask(input->nDims(), false);
-            Val* num_features = new Double(1);
-            for (size_t idx = 0; idx < kNormShapeNumDims; ++idx) {
-              const size_t axis = input->nDims() - 1 - idx;
-              inner_reduction_axes[idx] = axis;
-              inner_broadcast_mask[axis] = true;
-              num_features =
-                  mul(num_features, input->domain()->domain()[axis]->extent());
-            }
+              auto reciprocal_size = unaryOp(UnaryOpType::Reciprocal, N);
+              auto grad_in = mul(mul(reciprocal_size, bcast_rstd), inner);
+              value_map.emplace(node->output(0)->unique(), grad_in);
 
-            // TODO: NAN when mean and variance are zero
-            // --ftz=true -- flush-to-zero
-
-            // Algorithm
-            auto x_sum = sum(input, inner_reduction_axes);
-            auto x_sum_bcast = broadcast(x_sum, inner_broadcast_mask);
-            auto x_mean = div(x_sum_bcast, num_features);
-            auto x_mean_sub = sub(input, x_mean);
-            auto x_mean_sub_pow = mul(x_mean_sub, x_mean_sub);
-            auto var_sum = sum(x_mean_sub_pow, inner_reduction_axes);
-            auto var_sum_bcast = broadcast(var_sum, inner_broadcast_mask);
-            auto var = div(var_sum_bcast, num_features);
-            auto var_eps = add(var, new Double(kEps));
-            auto rvar = unaryOp(UnaryOpType::Rsqrt, var_eps);
-            auto output = mul(x_mean_sub, rvar);
-
-            // Optional: norm * weight
-            if (weight) {
-              auto weight_bcast = broadcast(weight, outer_broadcast_mask);
-              output = mul(output, weight_bcast);
+              if (output_mask[1]) {
+                auto grad_weight =
+                    sum(mul(grad_out, x_hat), outer_reduction_axes);
+                value_map.emplace(node->output(1)->unique(), grad_weight);
+              }
```

Comment:
github formatted this very strangely for me.
Anyway, we shouldn't leave dangling map when output_mask are false. You should still create/map an empty TensorView for the return tensor.